### DISCUSSION TITLE
Can Harfbuzz help mitigate an issue with Lao and Tibetan diacritics?

### DISCUSSION THREAD

**Discussion by zenodotus** [harfbuzz/harfbuzz#3107] [Q&A] [answered]
Hi @behdad and all,

I have been wondering whether there's a version of Harfbuzz that can help resolve an issue where Lao and Tibetan script diacritics are being positioned off of their base characters, as in the screenshot here:

![Lao Record Display](https://user-images.githubusercontent.com/15114916/128022770-5faccfaa-e9a0-486d-aecf-fa8c0d09e4dc.png)

The font being used in the CSS that displays this way is usually (but not necessarily always) Arial Unicode MS.  I'm seeing the diacritics positioned this way in two different versions of Chrome (91.0.4472.124 and 92.0.4515.107), using two different versions of Harfbuzz.

Best,
Charles Riley

**Comment by jfkthame** [comment, marked as answer]
Arial Unicode MS is fundamentally broken for scripts like this. It has basic glyphs, but no shaping support; and although it has a GDEF table, it doesn't even bother to classify the Lao vowel/tone marks as diacritics. So there's not really any chance of it working properly.

**Reply by zenodotus** [reply]
Thanks for your reply!

Charles

**Reply by behdad** [reply]
> Arial Unicode MS is fundamentally broken for scripts like this. It has basic glyphs, but no shaping support; and although it has a GDEF table, it doesn't even bother to classify the Lao vowel/tone marks as diacritics. So there's not really any chance of it working properly.

We actually deal with that for Hebrew, by disabling OTL and hence using fallback when there is no `hebr` script system in GPOS. Maybe we can do something similar in USE as well.

Note that that heuristic has also caused this: https://github.com/harfbuzz/harfbuzz/issues/3074
so maybe a better heuristic is desired; one detecting Arial Unicode MS specifically?

**Reply by jfkthame** [reply]
Fallback diacritic positioning would make things look a bit better (less obviously broken) at first glance, I suppose, but scripts like Tibetan and Lao would still have issues if the font doesn't actually have shaping support, because of things like virama handling for consonant clusters. I'm not sure it's worth trying to do anything with such a font.
